src/hooks/useCurrentRoute.ts

```diff
@@ -0,0 +1,35 @@
+import nextRoutes from '../'
+import { useRouter } from 'next/router'
+import { useMemo } from 'react'
+
+const getPathName = (path: string) => path.split(/(\?|#)/)[0]
+
+const useCurrentRoute = (routes: ReturnType<typeof nextRoutes>) => {
```

Comment:
we could make it as a factory so it could be created already in the context of routes so the signature would not change 


so we could have 
```
export const { Router, Link, useCurrentRoute } = routes
```
https://github.com/join-com/app/blob/73b33f1acb8a5a790f61f7800b5c13426809c485/src/server/routes.ts#L51

and in [`src/client/hooks/useCurrentRoute.ts`](https://github.com/join-com/app/blob/master/src/client/hooks/useCurrentRoute.ts)

just export it from server routes
`export { useCurrentRoute as default }  from 'server/routes'

to reduce changes in the app.

WDYT?

Reply:
Yeah, that would work as well but i don't see a point in extra useMemo 🤔 

Smt like that should be enough:
```suggestion
import routes from 'server/routes'

const useCurrentRoute = () =>  useCurrentRouteSharedCode(routes)
```

Reply:
Okay i see you point about factory pattern. Is something below good?

Extend and add public method `currentRoute` in [src/Routes.tsx](https://github.com/join-com/next-routes-with-locale/blob/master/src/Routes.tsx)

```typescript
   public currentRoute(asPath: string) {
      const getPathName = (path: string) => path.split(/(\?|#)/)[0]
      const pathName = getPathName(asPath)
      const route = this.findByPathname(pathName)
  
      const push = (
        params: Record<string, unknown>,
        {
          shallow = true,
        }: {
          shallow?: boolean
        } = {}
      ) => {
        const routeName = route?.name
  
        if (!routeName) {
          throw new Error('Missing route name')
        }
  
        return this.pushRoute(routeName, params, { shallow })
      }
  
      return { route, push } as const
  }
```

and below in [here](https://github.com/join-com/app/blob/73b33f1acb8a5a790f61f7800b5c13426809c485/src/server/routes.ts#L51)

```typescript
export const { Router, Link, currentRoute } = routes
```

and in [src/client/hooks/useCurrentRoute.ts](https://github.com/join-com/app/blob/master/src/client/hooks/useCurrentRoute.ts)

```typescript

import { useRouter } from 'next/router'

import { currentRoute } from 'server/routes'

const useCurrentRoute = () => {
    const { asPath, push, route } = useRouter()
    const route = currentRoute(asPath)

    return { route, push, routes } as const
  }


export default useCurrentRoute


```

Reply:
With this approach, you will trigger `this.findByPathname(pathName)` in each render so the hook will be more expensive.

I would stick with https://github.com/join-com/next-routes-with-locale/pull/17#discussion_r1357051686